dashboard/src/shared/ResourceRef.ts

```diff
@@ -39,6 +39,8 @@ class ResourceRef {
     switch (this.kind) {
```

Comment:
Nitpick but it might make sense to implement this as a map so extending it is simpler.

Reply:
Ah good shout, I agree that would be much easier.